tests/mod.nu

```diff
@@ -171,3 +169,211 @@ export def generate-local-registry [] {
+        assert equal $renamed_reg.url "https://example.com/test.nuon"
+
+        # Rename again
+        nupm registry rename renamed-registry final-name
+
+        let registries_final = nupm registry list
+        let final_reg = $registries_final | where name == "final-name" | first
+        assert equal $final_reg.url "https://example.com/test.nuon"
+        assert equal ($registries_final | where name == "renamed-registry" | length) 0
+    }
+}
+
+export def registry-describe [] {
+    with-test-env {
+        # Describe the test registry that's already configured
+        let description = nupm registry describe test
+
+        # Verify we get package information
+        assert (($description | length) > 0)
+
+        # Check for expected packages from the test registry
+        let spam_scripts = $description | where name == "spam_script"
+        assert (($spam_scripts | length) > 0)
+
+        # Check that we have the latest version
+        let spam_script_latest = $spam_scripts | where version == "0.2.0"
+        if (($spam_script_latest | length) > 0) {
+            let pkg = $spam_script_latest | first
+            assert equal $pkg.name "spam_script"
+            assert equal $pkg.source "local"
+            assert equal $pkg.version "0.2.0"
+        }
+
+        # Test error case with non-existent registry
+        let describe_result = try {
+            nupm registry describe non-existent-registry
+            "should not reach here"
+        } catch {|err|
+            $err.msg
+        }
+
+        assert ("Registry 'non-existent-registry' not found" in $describe_result)
+    }
+}
+
+export def registry-fetch [] {
+    with-test-env {
+        # Test fetch with local registry (test registry)
+        let fetch_result = try {
+            nupm registry fetch test
+            "success"
+        } catch {|err|
+            $err.msg
+        }
+
+        # For local registry, fetch should succeed
+        assert equal $fetch_result "success"
+
+        # Verify cache directory was created
+        let cache_dir = $env.nupm.cache | path join test
+        assert ($cache_dir | path exists)
+        assert (($cache_dir | path join "registry.nuon") | path exists)
+
+        # Verify package files were cached
+        let spam_script_cache = $cache_dir | path join "spam_script.nuon"
+        assert ($spam_script_cache | path exists)
+
+        # Test --all flag (only with local registries to avoid network issues)
+        let fetch_all_result = try {
+            nupm registry fetch --all
+            "success"
+        } catch {|err|
+            $err.msg
+        }
+
+        assert equal $fetch_all_result "success"
+
+        # Test error cases
+        let no_name_result = try {
+            nupm registry fetch
+            "should not reach here"
+        } catch {|err|
+            $err.msg
+        }
+
+        assert ("Please specify a registry name or use --all flag" in $no_name_result)
```

Comment:
I could use some suggestion on how to make this essentially do `(nupm registry fetch).is_err()`